src/js/main.js

```diff
@@ -66,13 +66,66 @@ function appReady() {
             },
         });
 
-        initializeSerialBackend();
+        checkSetupAnalytics(function (analyticsService) {
+            analyticsService.sendEvent(analyticsService.EVENT_CATEGORIES.APPLICATION, 'SelectedLanguage', i18n.selectedLanguage);
+        });
 
-        $('.connect_b a.connect').removeClass('disabled');
-        $('.firmware_b a.flash').removeClass('disabled');
+        initializeSerialBackend();
     });
 }
 
+function checkSetupAnalytics(callback) {
+    if (!analytics) {
+        setTimeout(function () {
+            const result = ConfigStorage.get(['userId', 'analyticsOptOut', 'checkForConfiguratorUnstableVersions' ]);
+            if (!analytics) {
+                setupAnalytics(result);
+            }
+
+            callback(analytics);
+        });
+    } else if (callback) {
+        callback(analytics);
+    }
+}
+
+function getBuildType() {
+    return GUI.Mode;
+}
+
+function setupAnalytics(result) {
+    let userId;
+    if (result.userId) {
+        userId = result.userId;
+    } else {
+        const uid = new ShortUniqueId();
+        userId = uid.randomUUID(13);
+
+        ConfigStorage.set({ 'userId': userId });
+    }
+
+    const optOut = !!result.analyticsOptOut;
+    const checkForDebugVersions = !!result.checkForConfiguratorUnstableVersions;
+
+    const debugMode = typeof process === "object" && process.versions['nw-flavor'] === 'sdk';
+
```

Comment:
If we adding this, can this at least be moved into separate file/package. `main.js` is for bootstrapping the app and this just adds noise to what's really is important here.